chore(C): add reverse and merge linked lists

rahulv07 opened the pull request:
<!--
Please make sure to read the Contribution Guidelines:
-->

<!-- PULL REQUEST TEMPLATE -->
<!-- (Update "[ ]" to "[x]" to check a box) -->

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] New algorithm
- [ ] Optimization in previous algorithms
- [ ] Code style update
- [ ] Refactor
- [ ] Documentation
- [ ] Other, please describe:

**Briefly describe the changes in this PR**

<!-- For example you can add algorithm name, language used to implement it and link to online implementation of it -->
I added two new algorithms using **C language**. 
1. Merge two linked-lists:
         This algorithm will merge two linked-lists. That is, it will append one linked-list to the end of another linked-list.
2. Reverse a linked-list:
         This algorithm will reverse the given linked-list. See this article for more info [Reverse a linked list](https://www.geeksforgeeks.org/reverse-a-linked-list/)

And I added my new algorithms in the README.md of the **C** folder.

**Other information:**
<!-- Add any additional info you want here -->

Arsenic-ATG commented on algorithms/C/linked-lists/merge-two-linkedlists.c:
```diff
@@ -0,0 +1,130 @@
+    //Getting values to enter into List B
+    printf("\nEnter the value for B: ");
+    scanf("%d",&b);
+    lb->data=b;
```
**Possible null dereference [[CWE-690]](https://cwe.mitre.org/data/definitions/690.html):** 

`lb` will point to `NULL` in case malloc() fails to allocate the memory ( either due to insufficient memory or memory corruption ) 

this is the case thought the source.
Always perform check after allocating memory dynamically to make sure you have a valid pointer there, you would rather terminate the program or handle this this case on your own instead of leaving a possible undefined behaviour in there.

**Fix:**
perform a check before dereferencing the pointer.

rahulv07 replied:
Yes sure. I will make changes in my code regarding this.

Arsenic-ATG commented on algorithms/C/linked-lists/merge-two-linkedlists.c:
```diff
@@ -0,0 +1,130 @@
+{
+    struct node *temp=la;
+
+    //Making the last node of List A , to point to the head node of List B
```
isn't it more like joining or concatenating the list instead of "merging" them as the order of the original lists are not reserved at the end ?

rahulv07 replied:
Yes. We can say like that. Its more like concatenating the lists.

Arsenic-ATG commented on algorithms/C/linked-lists/merge-two-linkedlists.c:
```diff
@@ -0,0 +1,130 @@
+    //Getting values to enter into List A
+    printf("\nEnter the value for A: ");
+    scanf("%d",&a);
+    la->data=a;
```
What will happen when you will pass 0 as the size of the number of nodes ?

rahulv07 replied:
I didn't think about that. Even if we pass 0, the program will ask for a value to enter into the list. Thank you for mentioning! I will change my code and solve this problem

rahulv07 commented:
I made further changes to my code. 

**Changes in merge-linkedlists.c:**
     1. Added validation for user input.
     2. Added a functionality to check whether memory is allocated or not
     3. The user can now specify the order of merging.
    
**Changes in reverse-linkedlists.c:**
     1. Added validation for user input
     2. Added a functionality to check whether memory is allocated or not

rahulv07 commented:
@Arsenic-ATG Please review my changes.

rahulv07 commented:
Thankyou👍

ming-tsai commented:
Hi @rahulv07,
Sorry I was occupied.

rahulv07 commented:
> Hi @rahulv07,
> Sorry I was occupied.

No problem. Thankyou for merging :)
